### ISSUE TITLE
runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by dayongi** [apache/apisix-ingress-controller#32]
**An error occurred in the Ingress controller when I added the Ingress configuration:**
```shell
[root@k8s-master ~]# kubectl logs -f ingress-controller-5587c86b49-5kz82 -n cloud
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x104bf0f]

goroutine 131 [running]:
github.com/gxthrj/seven/apisix.(*Route).convert(0xc000457c80, 0x0, 0x0, 0x1f045f8, 0xc0002e2390, 0x0)
        /go/pkg/mod/github.com/gxthrj/seven@v0.1.9/apisix/route.go:152 +0x1cf
github.com/gxthrj/seven/apisix.ListRoute(0x0, 0x0, 0x15c0580, 0xc0004c1150, 0x8, 0x10, 0xc0004c1120)
        /go/pkg/mod/github.com/gxthrj/seven@v0.1.9/apisix/route.go:54 +0x3c2
github.com/gxthrj/seven/apisix.FindCurrentRoute(0xc00024dc00, 0x13, 0xc0004c1120, 0x0)
        /go/pkg/mod/github.com/gxthrj/seven@v0.1.9/apisix/route.go:22 +0xe5
github.com/gxthrj/seven/state.(*routeWorker).trigger(0xc000401e60, 0x13f2fe9, 0x7, 0x13f269c, 0x6, 0x1274d40, 0xc0000ba550, 0x0, 0x0)
        /go/pkg/mod/github.com/gxthrj/seven@v0.1.9/state/builder.go:92 +0x1dd
github.com/gxthrj/seven/state.(*routeWorker).start.func1(0xc000401e60)
        /go/pkg/mod/github.com/gxthrj/seven@v0.1.9/state/route_worker.go:21 +0x8a
created by github.com/gxthrj/seven/state.(*routeWorker).start
        /go/pkg/mod/github.com/gxthrj/seven@v0.1.9/state/route_worker.go:17 +0x6c
```

**ingress configuration:** 
```yaml
kubectl apply -f - <<EOF
apiVersion: apisix.apache.org/v1
kind: ApisixRoute
metadata:
  name: foo-bar

spec:
  rules:
  - host: foo.bar
    http:
      paths:
      - backend:
          serviceName: http-svc
          servicePort: 80
        path: /
EOF
```

**K8S Service:**   I think the service is all right
```shell
[root@k8s-master eplat-yamls]# kubectl get svc -o wide --show-labels
NAME           TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)                         AGE     SELECTOR        LABELS
apisix-gw-lb   NodePort    10.1.41.185   <none>        9080:31700/TCP,9443:30905/TCP   20h     app=apisix-gw   app=apisix-gw
http-svc       NodePort    10.1.83.151   <none>        80:31562/TCP                    93m     app=http-svc    app=http-svc
kubernetes     ClusterIP   10.1.0.1      <none>        443/TCP                         3d17h   <none>          component=apiserver,provider=kubernetes

[root@k8s-master eplat-yamls]# curl http://10.1.83.151 -H Host:foo.bar


Hostname: http-svc-674f6fb5b5-gmh9m

Pod Information:
        node name:      k8s-node1
        pod name:       http-svc-674f6fb5b5-gmh9m
        pod namespace:  default
        pod IP: 10.244.1.68

Server values:
        server_version=nginx: 1.13.3 - lua: 10008

Request Information:
        client_address=10.244.0.0
        method=GET
        real path=/
        query=
        request_version=1.1
        request_scheme=http
        request_uri=http://foo.bar:8080/

Request Headers:
        accept=*/*
        host=foo.bar
        user-agent=curl/7.29.0

Request Body:
        -no body in request-
```

**cloud.yaml: **

> Maybe the `cloud.yaml` need modify, Because the `etcd` is a pod on the `master node` and must use https to connect it, It's possible the ingress controller not got data

Because the `etcd` is a pod on the `master node` ,so i  don't know the `ETCD_SERVER_INTERNAL`  what to put in here
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: cloud-config
  namespace: cloud
data:
  ETCD_SERVER_INTERNAL: '["http://127.0.0.1:2379"]'
  SYSLOG_HOST: 127.0.0.1
  APISIX_BASE_URL: "http://10.25.78.50:31700/apisix/admin/route/apisix/admin"
  ENV: "prod"
```

**Comment by gxthrj** [comment]
it seemed that your apisix has already defined routes before,and the `desc` is nil
two  solutions:
1. if the data is not important, you can remove all routes, and use CRD defined only;
2.or you can add `desc` field with CRD rules: `host` + `path`;

**Comment by tokers** [comment]
Duplicated with #14 .

**Comment by tokers** [comment]
Solved since #9 merged.
